### PR TITLE
feat: Hubspot plugin

### PR DESCRIPTION
**Pull request by shimonp21**
Leaving for future work:
- Add a "throttler"/"rate-limiter" so we don't hit the hubspot rate-limit so much (their rate-limit is quite low ~=150/10s). https://developers.hubspot.com/docs/api/usage-details#rate-limits.
- Related to above point:  No [associations](https://developers.hubspot.com/docs/api/crm/associations) resources were added, due to the fact that the API requires an API call `per-resource` and `per-resource-type`. i.e. to fetch all contact associations, it will require, **for every contact**, to make "List-Contact-Company-Associations", "List-Contact-Deal-Associations", "List-Contact-Ticket-Associations". i.e. ~10 API calls **per resource**! Skipping these for now, even though that sucks..
- Fix bugs in the https://github.com/clarkmcc/go-hubspot library for `cms` resources (none of those can be fetched with this library).
- Add `Engagement` resources (will require contributing to `clarkmcc` library).

### CODE REVIEW

File: plugins/source/hubspot/.github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: release
```
Comment (erezrokah) [inline plugins/source/hubspot/.github/workflows/release.yaml:1]: Is this file needed?

File: website/pages/docs/plugins/sources/hubspot/overview.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { getLatestVersion } from "../../../../../utils/versions";
+import { Badge } from "../../../../../components/Badge";
+import { Callout } from 'nextra-theme-docs';
+
+<Badge text={"Latest: " + getLatestVersion("source", "hubspot")}/>
+
+The CloudQuery HubSpot plugin extracts HubSpot information into your destination. It is based on the [HubSpot API](https://developers.hubspot.com/docs/api/overview) and the [github.com/clarkmcc/go-hubspot](https://github.com/clarkmcc/go-hubspot) library.
+
+<Callout type="warning">
+The HubSpot API is heavily [rate-limited](https://developers.hubspot.com/docs/api/usage-details#rate-limits). If you encounter
```
Comment (erezrokah) [inline website/pages/docs/plugins/sources/hubspot/overview.md:12]:
```suggestion
The HubSpot API is heavily [rate limited](https://developers.hubspot.com/docs/api/usage-details#rate-limits). If you encounter 
```

Comment (erezrokah) [inline website/pages/docs/plugins/sources/hubspot/overview.md:12]:
```suggestion
The HubSpot API is heavily [rate limited](https://developers.hubspot.com/docs/api/usage-details#rate-limits). If you encounter 
```

File: plugins/source/hubspot/go.mod [hunk 3]
```diff
@@ -0,0 +1,37 @@
+module github.com/cloudquery/cloudquery/plugins/source/hubspot
+
+go 1.19
+
+require (
+	github.com/clarkmcc/go-hubspot v0.0.0-20221010213350-20c2f9cbf936
+	github.com/cloudquery/plugin-sdk v1.25.0
```
Comment (erezrokah) [inline plugins/source/hubspot/go.mod:7]: Can we update to the latest SDK?
See https://github.com/cloudquery/plugin-sdk/releases/tag/v1.28.0

File: plugins/source/hubspot/client/client.go [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+		"Line_items",
+		"Meetings",
+		"Products",
+		"Feedback_submissions",
+	}
+
+	return &Client{
+		Logger:         logger,
+		Authorizer:     hubspot.NewTokenAuthorizer(authToken),
+		AllObjectTypes: objectTypes,
```
Comment (erezrokah) [inline plugins/source/hubspot/client/client.go:75]: Is `AllObjectTypes` used anywhere? I couldn't find where

File: plugins/source/hubspot/.github/workflows/release.yaml (hunk 1 above)
Comment (shimonp21) [reply plugins/source/hubspot/.github/workflows/release.yaml:1]: No idea what that is :)

Comment (erezrokah) [reply plugins/source/hubspot/.github/workflows/release.yaml]: Did you use the scaffold CLI for the plugin? If so, it comes from https://github.com/cloudquery/cloudquery/tree/d75df29d468a84724229ecef6ce558d31e665707/scaffold/cmd/templates/source/.github/workflows, but it is not needed for the monorepo, so we can delete it

File: plugins/source/hubspot/client/client.go (hunk 4 above)
Comment (shimonp21) [reply plugins/source/hubspot/client/client.go]: Yep, leftovers from deleted code :)

**Comment by shimonp21** [comment]
Deleted Codegen :)

File: plugins/source/hubspot/resources/services/crm/companies.go [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+import (
+	"github.com/cloudquery/plugin-sdk/schema"
+)
+
+func Companies() *schema.Table {
+	return &schema.Table{
+		Name:        "hubspot_crm_companies",
+		Description: "https://developers.hubspot.com/docs/api/crm/companies",
+		Resolver:    fetchCompanies,
```
Comment (erezrokah) [inline plugins/source/hubspot/resources/services/crm/companies.go:11]: Can we change the tables to use:
https://github.com/cloudquery/plugin-sdk/blob/281e706260eb83e70992e672064e4f35eb51ab4b/transformers/struct.go#L116

It would mean less code and also if an SDK update adds a new field we won't need to manually add it.

Also it has a `WithPrimaryKeys` option so you don't need to redefine the `id` column

File: website/pages/docs/plugins/sources/hubspot/overview.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import { Callout } from 'nextra-theme-docs';
+
+<Badge text={"Latest: " + getLatestVersion("source", "hubspot")}/>
+
+The CloudQuery HubSpot plugin extracts HubSpot information into your destination. It is based on the [HubSpot API](https://developers.hubspot.com/docs/api/overview) and the [github.com/clarkmcc/go-hubspot](https://github.com/clarkmcc/go-hubspot) library.
+
+<Callout type="warning">
+The HubSpot API is heavily [rate limited](https://developers.hubspot.com/docs/api/usage-details#rate-limits). If you encounter 
+rate limit errors when running the HubSpot plugin, please let us know by opening an issue on our [GitHub repository](https://github.com/cloudquery/cloudquery) or by contacting us on [Discord](https://www.cloudquery.io/discord).
+</Callout>
```
Comment (erezrokah) [inline website/pages/docs/plugins/sources/hubspot/overview.md:14]:
```suggestion
<Callout type="warning">

The HubSpot API is heavily [rate limited](https://developers.hubspot.com/docs/api/usage-details#rate-limits). If you encounter 
rate limit errors when running the HubSpot plugin, please let us know by opening an issue on our [GitHub repository](https://github.com/cloudquery/cloudquery) or by contacting us on [Discord](https://www.cloudquery.io/discord).

</Callout>
```

For some reason not having spaces confuses our linter

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks for the follow up @shimonp21.
I think there are still some leftovers from the code gen (tables not using the transformer)

**Review by erezrokah — APPROVED** [review]
:shipit:
